pages/profile/[id].tsx

```diff
@@ -0,0 +1,114 @@
+    return name.endsWith('s') ? `${name}'` : `${name}'s`;
+  }
+
+  return (
+    <>
+      <Container className="flex flex-col gap-4 pt-20 items-center">
+        <Box component="header">
+          <Typography variant="h1" className="text-5xl font-700 text-center">
+            {possessiveForm(user.discord_name)} Profile
+          </Typography>
+        </Box>
+
+        {user.bio && <Typography variant="body1" component="p">{user.bio}</Typography>}
+
+        <List className="flex gap-2">
+          {user.twitter_username && (
+            <ListItemButton
+              component="a"
+              href={`https://twitter.com/${user.twitter_username}`}
+              target="_blank"
+              rel="noopener noreferrer"
+              className="p-2 rounded-full bg-gray-100 hover:bg-gray-200"
+            >
+              <ListItemIcon sx={{ minWidth: 'auto' }}>
+                <TwitterIcon className="text-gray-800"/>
+              </ListItemIcon>
+            </ListItemButton>
+          )}
+
+          {user.linkedin_url && (
+            <ListItemButton
+              component="a"
+              href={user.linkedin_url}
+              target="_blank"
+              rel="noopener noreferrer"
+              className="p-2 rounded-full bg-gray-100 hover:bg-gray-200"
+            >
+              <ListItemIcon sx={{ minWidth: 'auto' }}>
+                <LinkedInIcon className="text-gray-800"/>
+              </ListItemIcon>
+            </ListItemButton>
+          )}
+
+          {user.github_username && (
+            <ListItemButton
+              component="a"
+              href={`https://github.com/${user.github_username}`}
+              target="_blank"
+              rel="noopener noreferrer"
+              className="p-2 rounded-full bg-gray-100 hover:bg-gray-200"
+            >
+              <ListItemIcon sx={{ minWidth: 'auto' }}>
+                <GitHubIcon className="text-gray-800"/>
+              </ListItemIcon>
+            </ListItemButton>
+          )}
+
+          {user.website && (
+            <ListItemButton
+              component="a"
+              href={user.website}
+              target="_blank"
+              rel="noopener noreferrer"
+              className="p-2 rounded-full bg-gray-100 hover:bg-gray-200"
+            >
+              <ListItemIcon sx={{ minWidth: 'auto' }}>
+                <HomeRoundedIcon className="text-gray-800"/>
+              </ListItemIcon>
+            </ListItemButton>
+          )}
+        </List>
+      </Container>
+    </>
+  );
+};
+
+export const getServerSideProps: GetServerSideProps<Props> = async ({ req, query }) => {
+  const { origin } = absoluteUrl(req);
+
+  try {
+    const config: RawAxiosRequestConfig = req ? {
+      withCredentials: true,
+      headers: {
+        cookie: req.headers.cookie,
+      },
+    } : {};
+
+    const { data: user } = await axios.get<UserProfile>(`${origin}/api/users/${query.id}`, config);
```

Comment:
how do you feel about doing a minor refactor here - pulling this out into a client lib file for making requests - so that every API request doesn't need to re-define the axios config? :) totally optional, i get if you'd rather it be a separate task (that either you or anyone else can take on)

Reply:
I actually made a note about that and was going to create an issue or do it myself later in a separate PR, but yea... I can throw it in. 👍🏼